pkgs/net-lib/net/server.rkt

```diff
@@ -91,12 +91,17 @@
     (thread-wait server-thd)))
 
 (define run-server
-  (procedure-rename
-   (make-keyword-procedure
-    (lambda (kws kw-args host port handle)
-      (parameterize-break #f
-        (define stop (keyword-apply start-server kws kw-args host port handle null))
-        (with-handlers ([exn:break? void])
-          (sync/enable-break never-evt))
-        (stop))))
-   'run-server))
+  (let-values ([(required-kws optional-kws) (procedure-keywords start-server)])
+    (procedure-rename
+     (procedure-reduce-keyword-arity
```

Comment:
Since you're using `procedure-reduce-keyword-arity` anyway, you don't need `procedure-rename`: you can just pass the name to `procedure-reduce-keyword-arity`. Also, it might be slightly more efficient to use `procedure-reduce-keyword-arity-mask` and `procedure-arity-mask`.